### ISSUE TITLE
conversations.json - has anyone developed bigger corpus?

### ISSUE THREAD

**Issue by antfellow** [gunthercox/ChatterBot#214]
Hi there  I have just downloaded the chatterbot.  Just wondering if anyone has developed any additional/bigger training corpus files? Thanks.

**Comment by gunthercox** [comment]
I'm aware that there are a few user-created ones floating around out there
for non-english languages. If you or anyone else is interested in
contributing a larger selection of corpus data to the project the addition
would be greatly welcomed.

**Comment by buaahsh** [comment]
hi antfellow, gunthercox, may you can try the [Ubuntu dialog dataset] (https://github.com/rkadlec/ubuntu-ranking-dataset-creator) , it's one of the largest public dialog datasets available, but it's not a chit-chat corpus.

hi @gunthercox , i have a question about the logic of statement. For a query-response pair, you used the response as the text property of the statement class, why not use the query as the text? Using the query as key, for the closest match function, we can directly get the closest key.

**Comment by z4nD4R** [comment]
Hi all,

I have tried to create a larger corpus in my own language. I've succesfully created a conversations.json corpus file. The file itself is ~16MB big. When I tried to to use 
```chatterbot.train("chatterbot.corpus.slovak")``` after 30min the import/train will crash with `Number of recursion error`. Anyone experienced this issue?

**Comment by sumn2u** [comment]
Maybe we can try it with hadoop MapReduce @gunthercox  what you say  ?

**Comment by davizucon** [comment]
Maybe we can use Redis as storage adapter.

**Comment by kizivat** [comment]
Hi, @z4nD4R. 
I've just created a script to extract dialogues from [OPUS corpus](http://opus.nlpl.eu/OpenSubtitles2018.php) (movies subtitles). Currently, it's the repository is private but if there would be an interest I can publish it after some polishing. Being Slovak myself I also already create a corpus as big as 600MB including over 180k conversations. That's after excluding sentences containing profanity words (and there was a lot of them, given it's from movies...). The profanity filter is not included in the repo, but I could share with you the Slovak one I created.
